packages/suite/src/components/suite/modals/Pin/index.tsx

```diff
@@ -2,11 +2,20 @@ import React from 'react';
 import styled from 'styled-components';
 import { Translation, Modal, ModalProps, PinMatrix } from '@suite-components';
 import { TrezorDevice } from '@suite-types';
+import { variables } from '@trezor/components';
 
 const StyledModal = styled(Modal)`
     width: unset;
 `;
 
+// limit width to prevent extending the modal past the width of the pin matrix
+// setting max width always to fit-content would cause shrinking the pin matrix to its min width and creating a linebreak in the description
+const Wrapper = styled.div<{ isExtended: boolean }>`
```

Comment:
What about styling `Description` in the `Modal` itself so that all similar cases are covered? `'fit-content'` should do the job?

Reply:
Yeah, I tried to come up with a general solution, but did not find anything satisfying. Setting `max-width: fit-content` on every `Description` might lead to undesired shrinking of the modal when there is no other wide content.

![Screenshot 2022-04-12 at 13 12 23](https://user-images.githubusercontent.com/42465546/162958102-dcc725e9-fd77-4a68-86a8-f263925f7587.png)

Reply:
Imo, can be fixed by having some `min-width` in `Modal` 🤔 but also I feel like having no content is an unrealistic case. @tomasklim what do you think?

Reply:
`min-width` makes sense to me. However, it has to be different for different screen sizes. I don't see a use case for an empty modal 😄

Reply:
@komret In that case, let's proceed with a general solution, please 🤓

Reply:
I kept separate logic for the `Pin` modal to keep the `Description` on one line when possible, but also added `max-width: fit-content` to the `Modal` component for general use.

Modals generally have a set `width`, so `min-width` would only come into play when `width` is overridden AND there is no other content than description. I consider the current solution sufficient and it does not introduce arbitrary numbers, but let me know if you want me to set it anyway.